### PR TITLE
CDK Init hello-cdk

### PR DESCRIPTION
**Pull request by russomi**
* Adds key concepts and diagram
* Lint using [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint)
* Reformatting using [markdown-formatter](https://marketplace.visualstudio.com/items?itemName=mervin.markdown-formatter)
* Completes steps for hello-cdk quick start

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,7 +14,7 @@ dist/
 downloads/
 eggs/
 .eggs/
-lib/
+#lib/
 lib64/
```
Comment (russomi) [inline .gitignore:18]: This entry prevented adding the `lib/` directory for local modules.

File: .gitignore [hunk 2]
```diff
@@ -253,3 +253,6 @@ package-lock.json
 # CDK asset staging directory
 .cdk.staging
 cdk.out
+
+# vscode
+.vscode
```
Comment (russomi) [inline .gitignore:258]: Ignore .vscode folder?

File: README.md [hunk 3]
```diff
@@ -1,31 +1,120 @@
+# aws-cdk-python
+
 ![https://cdkworkshop.com/images/cdk-logo.png](https://cdkworkshop.com/images/cdk-logo.png)
 
-# aws-cdk-python
+Based on [Your first AWS CDK app](https://docs.aws.amazon.com/cdk/latest/guide/hello_world.html).
```
Comment (russomi) [inline README.md:5]: markdownlint error:  first line must be `# title`

File: README.md [hunk 4]
```diff
@@ -1,31 +1,120 @@
...
+* [Background](#background)
+* [Key Concepts](#key-concepts)
+* [Prerequisites](#prerequisites)
+* [Installation](#installation)
+* [Quick Start](#quick-start)
+* [Next steps](#next-steps)
+* [Contributing](#contributing)
+* [References](#references)
+* [License](#license)
+
```
Comment (russomi) [inline README.md:18]: Added table of contents with links to heading anchors

File: README.md [hunk 5]
```diff
@@ -1,31 +1,120 @@
...
+The AWS CDK is designed around a handful of important concepts. We will introduce a few of these here briefly. Follow the links to learn more, or see the Concepts topics in this guide's Table of Contents.
+
+An AWS CDK app is an application written in TypeScript, JavaScript, Python, Java, or C# that uses the AWS CDK to define AWS infrastructure.
+<!-- markdownlint-disable MD004 MD009-->
+
+* `App` - An app defines one or more `stacks` 
+  + `Stacks` (equivalent to AWS CloudFormation stacks) contain `constructs` 
+    - `Construct` - defines one or more concrete `AWS resources` 
+      * `AWS Resource` - concrete resource defined in `construct` 
+
```
Comment (russomi) [inline README.md:38]: This hierarchy of objects is key 🔑

File: README.md [hunk 6]
```diff
@@ -1,31 +1,120 @@
...
 sam --version
+
+# Upgrade the aws-sam-cli
 brew upgrade aws-sam-cli
 ```
 
-### AWS Toolkit for Visual Studio
-
-Install the [AWS Toolkit for Visual Studio Code](https://aws.amazon.com/visualstudiocode/) which is an open-source plug-in for Visual Studio Code that makes it easier to create, debug, and deploy applications on AWS. The toolkit provides an integrated experience for developing AWS CDK applications, including the AWS CDK Explorer feature to list your AWS CDK projects and browse the various components of the CDK application.
+See [AWS CDK tools > SAM CLI](https://docs.aws.amazon.com/cdk/latest/guide/tools.html#sam) for more information.
```
Comment (russomi) [inline README.md:117]: relocated the `### AWS Toolkit for Visual Studio` section to flow better.

File: README.md [hunk 7]
```diff
@@ -55,10 +144,10 @@ cdk init app --language python
 python -m venv .env
 
 # activate the virtualenv
-source env/bin/activate
+source .env/bin/activate
 
 # install the packages from requirements.txt
-pip install -r requirements.txt
+pip3 install -r requirements.txt
```
Comment (russomi) [inline README.md:150]: I had issues with my local install if I didn't use `pip3` ... it may be a local path/alias issue.

File: README.md [hunk 8]
```diff
@@ -67,28 +156,33 @@ cdk ls
...
+class HelloCdkStack(core.Stack):
+
+    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        # The code that defines your stack goes here
+        s3.Bucket(self,
+                  "MyFirstBucket",
+                  versioned=True,)
 ```
```
Comment (russomi) [inline README.md:189]: incorporated default stack class

File: README.md [hunk 9]
```diff
@@ -150,15 +251,46 @@ As with cdk synth, you don't need to specify the name of the stack since there's
...
+# Generate a difference report between local vs. deployed
+cdk diff
+```
+
+The AWS CDK Toolkit queries your AWS account for the current AWS CloudFormation template for the hello-cdk stack, and compares it with the template it synthesized from your app. The Resources section of the output should look like the following.
+
+``` bash
+# Deploys the stack using AWS credentials
+cdk deploy
+```
```
Comment (russomi) [inline README.md:279]: Added `Modifying the app` section.

File: README.md [hunk 10]
```diff
@@ -150,15 +251,46 @@ As with cdk synth, you don't need to specify the name of the stack since there's
...
 
-TODO
+Now that you're done with the quick tour, destroy your app's resources to avoid incurring any costs from the bucket you created, as follows.
+
+``` bash
+# Destroy stack resources
+cdk destroy
+```
+
+**Note**: If cdk destroy fails, it probably means you put something in your Amazon S3 bucket. AWS CloudFormation won't delete buckets with files in them. Delete the files and try again.
```
Comment (russomi) [inline README.md:290]: Added `Destroying apps resources` section.

File: README.md [hunk 11]
```diff
@@ -150,15 +251,46 @@ As with cdk synth, you don't need to specify the name of the stack since there's
...
 
 ## Next steps
 
-TODO
+* Try the [CDK Workshop](https://cdkworkshop.com/) for a more in-depth tour involving a more complex project.
+* See the [API reference](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-construct-library.html) to begin exploring the CDK constructs available for your favorite AWS services.
+* Dig deeper into concepts like [Environments](https://docs.aws.amazon.com/cdk/latest/guide/environments.html), [Assets](https://docs.aws.amazon.com/cdk/latest/guide/assets.html), [Permissions](https://docs.aws.amazon.com/cdk/latest/guide/permissions.html), [Runtime context](https://docs.aws.amazon.com/cdk/latest/guide/context.html), [Parameters](https://docs.aws.amazon.com/cdk/latest/guide/parameters.html), and [Escape hatches](https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html).
+* Explore [Examples](https://github.com/aws-samples/aws-cdk-examples) of using the AWS CDK.
```
Comment (russomi) [inline README.md:297]: Added `Next Steps` section

File: app.py [hunk 12]
```diff
@@ -1,11 +0,0 @@
...
-
-from aws_cdk import core
-
-from lib.HSCDKstack import HSCDKStack
-
-
-app = core.App()
-HSCDKStack(app, "hscdkstack")
-
-app.synth()
```
Comment (russomi) [inline app.py:11]: moved `app.py` into `hello-cdk` directory.

File: hello-cdk/setup.py [hunk 13]
```diff
@@ -15,11 +15,12 @@
...
 
-    package_dir={"": "tmp1"},
-    packages=setuptools.find_packages(where="tmp1"),
+    package_dir={"": "hello_cdk"},
+    packages=setuptools.find_packages(where="hello_cdk"),
 
     install_requires=[
-        "aws-cdk.core==1.45.0",
+        "aws-cdk.core==1.46.0",
+        "aws-cdk.aws-s3==1.46.0"
```
Comment (russomi) [inline hello-cdk/setup.py]: added `aws-cdk.aws-s3` and pinned version.

File: README.md (hunk 7 above)
Comment (ConorWright) [reply README.md:150]: I have pip3 aliased as pip. It saves a lot of hassle.

**Pull request closed without merging**